### PR TITLE
DEC-1156: Create audio video models

### PR DESCRIPTION
**Pull request by jgondron**
This adds the Audio and Video models. I ended up making our own version of enum that works with store_accessors now that we have 3 different entities each with their own state system. It works just like rails enum, except you can pass a store accessor so that the enum value will be stored in that store instead of on the base of the model.

### CODE REVIEW

**Comment by jgondron** [comment]
Please ignore the hound issues on the store enum file. This was mostly a copy/paste of rails enum and I'd rather not drift too far from the original just to fix style guide issues.

File: app/models/video.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require "store_enum"
+
+class Video < Media
+  extend StoreEnum
+  store_enum :data, status: { allocated: 0, ready: 1 }
+  store_accessor :data, :json_response
```
Comment (rfox2) [inline app/models/video.rb]: This will contain the thumbnail URI - just an fyi

File: lib/store_enum.rb [hunk 2]
```diff
@@ -0,0 +1,197 @@
+require 'active_support/core_ext/object/deep_dup'
+
```
Comment (rfox2) [inline lib/store_enum.rb]: If we're going to assume this library into our code base, I'd recommend bringing it up to our style standards.

File: lib/store_enum.rb [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+# Declare an enum attribute where the values map to integers in the database,
```
Comment (rfox2) [inline lib/store_enum.rb]: Can these comments be abbreviated?

**Comment by rfox2** [comment]
I didn't see that the enum code was a copy/paste from rails. I was suggesting we bring that up to our code standards if we're adopting that code into our base - most are pretty easy fixes. Apart from that, after the other Hound issues are fixed than 👍

File: app/models/media.rb [hunk 4]
```diff
@@ -2,4 +2,6 @@ class Media < ActiveRecord::Base
   belongs_to :collection
   has_many :items
   validates :collection, presence: true
+
+  has_paper_trail
```
Comment (jonhartzler) [inline app/models/media.rb]: Is there a spec for this ?

Comment (jgondron) [inline app/models/media.rb]: I have a "has a papertrail" inside of media_spec.rb

**Comment by jonhartzler** [comment]
👍

**Pull request merged**